autopush-common/src/reliability.rs

```diff
@@ -26,75 +26,32 @@ const CONNECTION_EXPIRATION: Duration = Duration::from_secs(10);
-    #[serde(rename = "accepted_webpush")]
-    IntAccepted, // Message was accepted by autoconnect from autopendpoint
-    #[serde(rename = "transmitted")]
-    Transmitted, // Message was handed off for delivery to the UA
-    #[serde(rename = "accepted")]
-    Accepted, // Message was accepted for delivery by the UA
-    #[serde(rename = "delivered")]
-    Delivered, // Message was provided to the WebApp recipient by the UA
-    #[serde(rename = "expired")]
-    Expired, // Message expired naturally (e.g. TTL=0)
-}
-
 // TODO: Differentiate between "transmitted via webpush" and "transmitted via bridge"?
-impl std::fmt::Display for ReliabilityState {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str(match self {
-            Self::Received => "received",
-            Self::Stored => "stored",
-            Self::Retrieved => "retrieved",
-            Self::Transmitted => "transmitted",
-            Self::IntTransmitted => "transmitted_webpush",
-            Self::IntAccepted => "accepted_webpush",
-            Self::Accepted => "accepted",
-            Self::Delivered => "delivered",
-            Self::Expired => "expired",
-        })
-    }
-}
-
-impl std::str::FromStr for ReliabilityState {
-    type Err = ApcError;
-
-    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
-        Ok(match s.to_lowercase().as_str() {
-            "received" => Self::Received,
-            "stored" => Self::Stored,
-            "retrieved" => Self::Retrieved,
-            "transmitted" => Self::Transmitted,
-            "accepted" => Self::Accepted,
-            "transmitted_webpush" => Self::IntTransmitted,
-            "accepted_webpush" => Self::IntAccepted,
-            "delivered" => Self::Delivered,
-            "expired" => Self::Expired,
-            _ => {
-                return Err(
-                    ApcErrorKind::GeneralError(format!("Unknown tracker state \"{}\"", s)).into(),
-                );
-            }
-        })
-    }
-}
-
-impl serde::Serialize for ReliabilityState {
-    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        serializer.serialize_str(&self.to_string())
-    }
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    serde::Deserialize,
+    serde::Serialize,
+    strum::Display,
+    strum::EnumString,
+)]
+#[serde(rename_all = "snake_case")]
+#[strum(serialize_all = "snake_case")]
+pub enum ReliabilityState {
+    Received,        // Message was received by the Push Server
+    Stored,          // Message was stored because it could not be delivered immediately
+    Retrieved,       // Message was taken from storage for delivery
+    IntTransmitted,  // Message was handed off between autoendpoint and autoconnect
+    IntAccepted,     // Message was accepted by autoconnect from autopendpoint
```

Comment:
@jrconlin I assume the previous, mismatched serialized names for these were a typo/aren't necessary, but if they're still needed we can restore them easily enough with one off procmacros on these fields

Reply:
wfm. So long as their differentiated, I don't really care.